tests/pytests/unit/utils/test_gitfs_locks.py

```diff
@@ -0,0 +1,613 @@
+"""
+These only test the provider selection and verification logic, they do not init
+any remotes.
+"""
+
+import logging
+import os
+import pathlib
+import signal
+import time
+
+import pytest
+from saltfactories.utils import random_string
+
+import salt.ext.tornado.ioloop
+import salt.fileserver.gitfs
+import salt.utils.files
+import salt.utils.gitfs
+import salt.utils.path
+import salt.utils.platform
+import salt.utils.process
+from salt.utils.immutabletypes import freeze
+from salt.utils.verify import verify_env
+
+try:
+    import pwd
+except ImportError:
+    import salt.utils.win_functions
+
+pytestmark = [pytest.mark.skip_unless_on_linux]
+
+log = logging.getLogger(__name__)
+
+
+def _get_user():
+    """
+    Get the user associated with the current process.
+    """
+    if salt.utils.platform.is_windows():
+        return salt.utils.win_functions.get_current_user(with_domain=False)
+    return pwd.getpwuid(os.getuid())[0]
+
+
+def _clear_instance_map():
+    try:
+        del salt.utils.gitfs.GitFS.instance_map[
+            salt.ext.tornado.ioloop.IOLoop.current()
+        ]
+    except KeyError:
+        pass
+
+
+class AdaptedConfigurationTestCaseMixin:
```

Comment:
You don't need to bring this into pytest...

Reply:
can you elaborate,  the pass ? or the class etc and reasons why ?

Reply:
Everything that is provided by the class is already provided as pytest fixtures, if not, fixtures can be created and should be preferred.

Reply:
Given the pita it was to get this working, I am reluctant to spend more time refactoring this when it works and have other tasks to get done.  This has been a pita to fix, and now that it is working, want to move on.
Have week before gone for two weeks, and want to get this is in before I go.